src/components/promoCode/services/promoCodeSrv.service.js

```diff
@@ -26,7 +33,7 @@
                 promoCodeStatusText[INVALID] = INVALID;
 
                 promoCodeSrv.checkPromoCode = function (promoCode, appContext) {
-                    var firebaseAppScopeName =  backendData[appContext].firebaseAppScopeName;
+                    var firebaseAppScopeName = backendData[appContext].firebaseAppScopeName;
```

Comment:
You set:
appContext = ENV.firebaseAppScopeName

So where is backendData[appContext].firebaseAppScopeName initialized?
And why do you have both:
backendData[appContext].firebaseAppScopeName
AND
appContext = ENV.firebaseAppScopeName